### ISSUE TITLE
Etcd advertisedSubnets

### ISSUE THREAD

**Issue by sergelogvinov** [siderolabs/talos#7947]
### Description

Etcd advertisedSubnets is the network (filtered ip) which can not exist on the node.
But we use https://github.com/siderolabs/talos/blob/e9c7ac17a9b707950b249e08e11ed7ddac64e8ae/internal/app/machined/pkg/controllers/etcd/spec.go#L97 NodeAddress (routed) to find the right IP.

Probably we need to use at least nodeIP+platformIP (dnat-ip). 
Or check if advertisedSubnets is a single IP use it as is?
What do you think?

Thanks.

### Environment

- Talos version: [1.5.x]
- Kubernetes version:
- Platform: azure

**Comment by smira** [comment]
Yes, probably advertised subnets might come from a more broad range of addresses. Looking for external IP as advertised? It might help if you could describe a scenario.

**Comment by sergelogvinov** [comment]
Use case - multi cloud setup without kubespan/ipsec/etc (direct connection).

To make p2p connections between etcd, we need to advertise the external (public) IP, which not exist on the node.
Currently I've added the IP alias to the interface...

PS. I did not notice any error messages when Talos did not find the advertised IP too. Probably we need to add error messages at boot time too.

Thanks.
